### DISCUSSION TITLE
How to access params IF app is at the route outside of Outlet?

### DISCUSSION THREAD

**Discussion by danieljmann96** [TanStack/router#955] [Q&A] [answered]
Wondered if I could get some advice on how to set this app up.

So in my RootRoute I have the following setup.
I have a search page route that handles its own params.

```
<VariousProviders>
    <NavBar />
    <Outlet />
</VariousProviders>
```

The NavBar is as follows. When the search is submitted, the navigation to the search page works great with the params being set correctly. I want to autofill the search bar to get the search query from the params, but the page won't always be on "/search".
```
const { searchQuery } = useSearch({ from: searchRoute.id });
return (
    <VariousStuff />
    <SearchBar defaultValue={searchQuery} />
)
```
If I load the page on any other route, the error below appears

> Invariant failed: useMatch("/search") is being called in a component that is meant to render the '__root__' route. Did you mean to 'useMatch("/search", { strict: false })' or 'useRoute("/search")' instead?

Is there a best practice way of accessing params IF the route is "/search", without throwing an error?
If this isn't super clear happy to provide more details.

For reference the search route is as below

```
const searchRoute = new Route({
    getParentRoute: () => appRoot,
    path: '/search',
    component: lazyRouteComponent(() => import('../components/SearchPage'),
    validateSearch: (
        search: Partial<SearchUrl> & SearchSchemaInput
    ): SearchUrl | Record<string, never> => {
        const result = SearchPageUrlParamParser.safeParse(search);
        if (result.success) {
            return result.data
        }
        return defaultSearchValues;
    }
});
```

**Comment by danieljmann96** [comment, marked as answer]
I answered the question myself, but I will leave the answer below:

In my navigation bar, I now do the following:

```
const router = useRouter()
....
<SearchBar defaultValue={router.state.location.search.searchQuery ?? ''} />
```

The "router.state.location.search" will be a partial type of all of the different route's params combined; pretty nice 👍

**Reply by schiller-manuel** [reply]
Why not `useSearch({ strict: false})` (hinted at by the error message)?

**Reply by danieljmann96** [reply]
> Why not `useSearch({ strict: false})` (hinted at by the error message)?

That gives a type error; it does still work when on the search route but also crashes my app when not on the search route.
